assignments/prototypes.js

```diff
@@ -8,20 +8,89 @@
   Each constructor function has unique properites and methods that are defined in their block comments below:
 */
   
-/*
-  === GameObject ===
-  * createdAt
-  * dimensions
-  * destroy() // prototype method -> returns the string: 'Object was removed from the game.'
-*/
 
-/*
-  === CharacterStats ===
-  * hp
-  * name
-  * takeDamage() // prototype method -> returns the string '<object name> took damage.'
-  * should inherit destroy() from GameObject's prototype
-*/
+  // === GameObject ===
+
+
+
+  function GameObject(attributes) {
+    this.createdAt = attributes.createdAt;
+    this.dimensions = attributes.dimensions;
+
+  };
+
+  GameObject.prototype.destroy = function() {
+    return `${this.name} was removed from the game.`
+  };
+
+  
+  CharacterStats.prototype = Object.create(GameObject.prototype);
+
+  // * createdAt
+  // * dimensions
+  // * destroy() // prototype method -> returns the string: 'Object was removed from the game.'
+
+
+  function CharacterStats(chrStatsAttribs) {
+    GameObject.call(this, chrStatsAttribs);
+    this.hp = chrStatsAttribs.hp;
+    this.name = chrStatsAttribs.name;
+
+  };
+
+  CharacterStats.prototype.takeDamage = function() {
+    return `${this.name} took damage.`
+  };
+
+
+
+Humanoid.prototype = Object.create(CharacterStats.prototype);
+
+
+function Humanoid(humAttribs) {
+  CharacterStats.call(this, humAttribs);
+  this.faction = humAttribs.faction;
+  this.weapons = humAttribs.weapons;
+  this.language = humAttribs.language;
+
+};
+
+Humanoid.prototype.greet = function() {
+  return `${this.name} offers a greeting in ${this.language}`
+};
+
+
+Hero.prototype = Object.create(Humanoid.prototype);
+Villain.prototype = Object.create(Humanoid.prototype);
+
+
+function Hero(heroAttribs) {
+  Humanoid.call(this, heroAttribs);
+  this.power = heroAttribs.power;
+  this.weakness = heroAttribs.weakness;
+};
+
+Hero.prototype.damage = function() {
+  return ` His hitpoints are now: ` + (this.hp -1);
```

Comment:
This method doesn't isn't very sensible. The methods on an object are its abilities. The methods of an object define the "behavior" of the object. In my view, the `damage` method on the `Hero.prototype` object is currently damaging the hero object instance. The instantiated hero objects would be harming themselves when the initiate the damage action. It would see better for hero to damage a target villain object (passed to the method as an argument)